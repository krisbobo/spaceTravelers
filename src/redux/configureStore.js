import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rockets';
import missionReducer from './missions/missions';
import dragonReducer from './dragons/dragons';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    dragons: dragonReducer,
  },
});
export default store;
