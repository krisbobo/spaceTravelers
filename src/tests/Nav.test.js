import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import NavBar from '../components/Nav';

const mockStore = configureStore([]);

describe('Test snapshot component of Navbar', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rockets: [
        {
          name: 'Rockets',
        },
      ],
    });
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </BrowserRouter>,
    );
  });
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
