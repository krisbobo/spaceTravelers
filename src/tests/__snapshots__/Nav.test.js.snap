// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot component of Navbar should render with given state from Redux store 1`] = `
<nav>
  <div
    className="logo"
  >
     
    <span
      className="icon-button"
    >
      <img
        alt="logo"
        src="logo.png"
        style={
          Object {
            "height": "3.5em",
            "width": "3.5em",
          }
        }
      />
    </span>
    <span
      className="nav-brand"
    >
      Space Travelers' Hub
    </span>
  </div>
  <ul
    className="nav-links"
  >
    <li>
      <a
        className="nav-link "
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="nav-link "
        href="/dragons"
        onClick={[Function]}
      >
        Dragons
      </a>
    </li>
    <li>
      <a
        className="nav-link "
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className="nav-link "
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
